### PR TITLE
Increase unit test coverage for WooPay direct checkout flow

### PR DESCRIPTION
**Pull request by ricardo**
Fixes 2554-gh-Automattic/woopay

#### Changes proposed in this Pull Request

This PR aims to increase the PHP and JavaScript unit test coverage for the direct to WooPay checkout flow.

Please refer to 2595-gh-Automattic/woopay for a description of the tests added for each individual issue in the project's epic.

#### Testing instructions

- Run `npm run test:js` and ensure all JavaScript tests pass.
- Run `npm run test:php` and ensure all PHP tests pass.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8407` or branch name `dev/increase-woopay-direct-checkout-unit-test-coverage` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=dev/increase-woopay-direct-checkout-unit-test-coverage) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: aa95fc740e95bee97bebfdd2152a95a7a0af4698
- Build time: 2024-04-05 19:39:25 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by ricardo** [comment]
Assigning @cesarcosta99 as reviewer based on the WooPay PR round-robin assignment.

File: tests/unit/test-class-wc-payments-features.php [hunk 1]
```diff
@@ -145,111 +136,94 @@ public function test_are_payments_enabled_returns_false_when_flag_not_set() {
...
 
-	public function test_is_wcpay_frt_review_feature_active_returns_true() {
-		add_filter(
-			'pre_option_wcpay_frt_review_feature_active',
-			function ( $pre_option, $option, $default ) {
-				return '1';
-			},
-			10,
-			3
+	public function test_is_woopay_direct_checkout_enabled_returns_true() {
```
Comment (cesarcosta99) [inline tests/unit/test-class-wc-payments-features.php:184]: To get complete coverage on this we should have a `test_is_woopay_direct_checkout_enabled_returns_false_when_first_party_auth_is_disabled` test. Do we have a reason for not including it?

**Review by cesarcosta99 — CHANGES_REQUESTED** [review]
All tests passed 🎉

 Just a couple of points:
- It was mentioned in the test coverage that we've added `WC_Payments_WooPay_Direct_Checkout` tests but I haven't seen any. Did we decide not adding any tests for that class, or maybe missed it?
- I left a suggestion below about doing another test case for completeness.

File: tests/unit/test-class-wc-payments-features.php (hunk 1 above)
Comment (ricardo) [reply tests/unit/test-class-wc-payments-features.php:184]: Good call. Fixed in https://github.com/Automattic/woocommerce-payments/pull/8407/commits/aa95fc740e95bee97bebfdd2152a95a7a0af4698.

**Comment by ricardo** [comment]
> It was mentioned in the test coverage that we've added `WC_Payments_WooPay_Direct_Checkout` tests but I haven't seen any. Did we decide not adding any tests for that class, or maybe missed it?

@cesarcosta99, The description on 2595-gh-Automattic/woopay was actually outdated about this one. I listed this test as a TODO for myself, but later realized there weren't any meaningful tests I could write about that class since it basically just enqueues the direct checkout scripts.

> I left a suggestion below about doing another test case for completeness.

Good call! I have added the test and requested a re-review.

**Review by cesarcosta99 — APPROVED** [review]
LGTM and tests well :shipit:

**Pull request merged**